examples/by_feature/gradient_accumulation.py

```diff
@@ -110,7 +110,10 @@ def training_function(config, args):
     seed = int(config["seed"])
     batch_size = int(config["batch_size"])
     # New Code #
-    gradient_accumulation_steps = int(args.gradient_accumulation_steps)
+    if accelerator.distributed_type != DistributedType.TPU:
+        gradient_accumulation_steps = int(args.gradient_accumulation_steps)
+    else:
+        gradient_accumulation_steps = 1
```

Comment:
No, you should raise an error at the beginning if on TPU with a gradient accumulation steps argument passed > 1.